src/pods_distribution.py

```diff
@@ -0,0 +1,180 @@
+import json
+import os
+from typing import Dict, Optional, List
+
+import dpath.util
+import psutil
+
+
+class DiskUtils:
+    @staticmethod
+    def get_size(directory_path) -> int:
+        total_size = 0
+        for root, dirs, files in os.walk(directory_path):
+            if root == directory_path:  # skip procfs if it is present ({directory_path}/proc)
+                dirs[:] = [d for d in dirs if d != "proc"]
+
+            for f in files:
+                fp = os.path.join(root, f)
+                if not os.path.islink(fp):  # skip if it is symbolic link
+                    total_size += os.path.getsize(fp)
+
+        return total_size
+
+
+class Container:
+    container_dir: str
+    id: str
+
+    container_type: str
+    container_name: Optional[str]
+    pod_name: str
+    pod_namespace: str
+
+    disk_size: int
+
+    def __init__(self, container_dir: str):
+        self.container_dir = container_dir
+        self.id = os.path.basename(container_dir)
+
+        config = self._get_config()
+        self.container_type = dpath.util.get(config, "annotations/io.kubernetes.cri.container-type")
+        self.container_name = dpath.util.get(config, "annotations/io.kubernetes.cri.container-name", default=None)
+        self.pod_name = dpath.util.get(config, "annotations/io.kubernetes.cri.sandbox-name")
+        self.pod_namespace = dpath.util.get(config, "annotations/io.kubernetes.cri.sandbox-namespace")
+
+        self.disk_size = self._get_disk_size()
+
+    def _get_config(self):
+        config_file_path = os.path.join(self.container_dir, "config.json")
+        with open(config_file_path, "r") as config_file:
+            json_config = config_file.read()
+        return json.loads(json_config)
+
+    def _get_disk_size(self) -> int:
+        root_fs_dir = os.path.join(self.container_dir, "rootfs")
+        return DiskUtils.get_size(root_fs_dir)
+
+    def __repr__(self):
+        return f"Container(container_type={self.container_type}, " \
+               f"container_name={self.container_name}, disk_size={self.disk_size})"
+
+
+class Pod:
+    pod_name: str
```

Comment:
Same here: https://home.robusta.dev/blog/common-python-mistakes-mixing-up-instance-and-class-members/

Reply:
same as above